content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+
+For the present purpose, we do not use the vanilla RNN, but a variant of it known as long short-term memory (LSTM) network. This latter, which involves components called gates, is often preferred over its competitors due to its better stability with respect to gradient update (vanishing and exploding gradient).
+better stability with gradient update (vanishing and exploding gradient) by using gates 
+([to learn more about LSTMs](http://colah.github.io/posts/2015-08-Understanding-LSTMs/)). 
+
+Also, for now, let's simply use a single layer unidirectional LSTM. We will, later on, explore the use of more layers and bidirectional approach. 
+
+### Word Embeddings
+
+Since our data is text, we will use a well-known word encoding technique, word embeddings. Word embeddings are vectors
+representation of words. The main hypothesis is that words have a linear relation along with them. For example, the linear relation
+between the word `king` and `queen` is gender. So logically, if we remove the vector of `male` to `king` and add the vector
+of `female`, we should obtain the vector of `queen`. That been said; usually, those kind of representation are in dimension `300`, which
+make it impossible for humans to reason about. Still, the idea is there but in a larger dimension space.
+
+So our LSTM input and hidden state dimensions will be of the same size as the word vectors. 
+This size corresponds to the word embeddings dimension, which in our case will be the 
+[French pre trained](https://fasttext.cc/docs/en/crawl-vectors.html) fastText embeddings of dimension `300`. 
+
+### The Pytorch Model
+
+But first, let's import all the needed packages.
+
+```python
+import os
+import pickle
+
+import fasttext
+import fasttext.util
+import requests
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from poutyne import set_seeds
+from poutyne.framework import Experiment
+from torch.nn.functional import cross_entropy
+from torch.nn.utils.rnn import pad_packed_sequence, pack_padded_sequence, pad_sequence
+from torch.utils.data import DataLoader
+```
+
+So now, let's create a single (i.e. one layer) unidirectional LSTM wit `input_size` and `hidden_size` of `300`. We 
+will explore later on the effect of stacking more layers and of using a bidirectional approach.
+
+> See [here](https://discuss.pytorch.org/t/could-someone-explain-batch-first-true-in-lstm/15402) the explanation why we use the `batch_first` argument.
+
+```python
+dimension = 300
+num_layer = 1
+bidirectional = False
+
+lstm_network = nn.LSTM(input_size=dimension,
+                       hidden_size=dimension,
+                       num_layers=num_layer,
+                       bidirectional=bidirectional,
+                       batch_first=True)
+```
+
+
+## Fully-connected Layer
+Since the output of the LSTM network is of `300`, we will use a fully-connected layer to map this large dimension space into
```

Comment:
```suggestion
Since the output of the LSTM network is of dimension `300`, we will use a fully-connected layer to map this large dimension space into
```